packages/manager-react-components/src/hooks/datagrid/useResourcesV6.tsx

```diff
@@ -100,7 +99,7 @@ export function useResourcesV6({
 
     if (sortData && sorting) {
       const type = columns?.find((element) => element.id === sorting.id)?.type;
-      const sortedDatas = sortData?.sort((a, b) =>
+      const sortedDatas = sortData?.sort((a: any, b: any) =>
```

Comment:
Is it really needed ? Inference type should determine the type is `T`, no ?

Reply:
It creates a build problem (implicit `any`)

And the problem if you put T you would have to make sure that `sorting.id` is a property in T so it's a bit complicated to type correctly the internal implementation...

In the end, we just need to make sure that `sortData` and `flattenData` are typed correctly (`T[]`) when you use the hook, I think. The internal implementation of sorting etc. doesn't need to be typed exactly

Reply:
For information I see for avoid any type : 

```
// Welcome to the TypeScript Playground, this is a website
// which gives you a chance to write, share and learn TypeScript.

// You could think of it in three ways:
//
//  - A location to learn TypeScript where nothing can break
//  - A place to experiment with TypeScript syntax, and share the URLs with others
//  - A sandbox to experiment with different compiler features of TypeScript

const anExampleVariable = "Hello World"
console.log(anExampleVariable)

// To learn more about the language, click above in "Examples" or "What's New".
// Otherwise, get started by removing these comments and the world is your playground.

type unknownSupposedValidKey = string | number | symbol;
type unknownObject = { [key: unknownSupposedValidKey]: unknown };

const sortData: unknownObject[] = []

const typee = "toto"

const sorting: { id: unknown, desc: boolean } = { id: 21, desc: true }

function sortColumn(typee: string, a: unknown, b: unknown, desc: boolean): number { return 1}


const sortedDatas = sortData?.sort((a: unknownObject, b: unknownObject) => sortColumn(typee, a?.[sorting.id as unknownSupposedValidKey], b?.[sorting.id as unknownSupposedValidKey], sorting?.desc))
```

It's a little long but if you want fix it with a better readability I think it's a good way

I directly try this type on : https://www.typescriptlang.org/play/?#code/PTAEHUFMBsGMHsC2lQBd5oBYoCoE8AHSAZVgCcBLA1UABWgEM8BzM+AVwDsATAGiwoBnUENANQAd0gAjQRVSQAUCEmYKsTKGYUAbpGF4OY0BoadYKdJMoL+gzAzIoz3UNEiPOofEVKVqAHSKymAAmkYI7NCuqGqcANag8ABmIjQUXrFOKBJMggBcISGgoAC0oACCbvCwDKgU8JkY7p7ehCTkVDQS2E6gnPCxGcwmZqDSTgzxxWWVoASMFmgYkAAeRJTInN3ymj4d-jSCeNsMq-wuoPaOltigAKoASgAywhK7SbGQZIIz5VWCFzSeCrZagNYbChbHaxUDcCjJZLfSDbExIAgUdxkUBIursJzCFJtXydajBBCcQQ0MwAUVWDEQC0gADVHBQGNJ3KAALygABEAAkYNAMOB4GRonzFBTBPB3AERcwABS0+mM9ysygc9wASmCKhwzQ8ZC8iHFzmB7BoXzcZmY7AYzEg-Fg0HUiQ58D0Ii8fLpDKZgj5SWxfPADlQAHJhAA5SASPlBFQAeS+ZHegmdWkgR1QjgUrmkeFATjNOmGWH0KAQiGhwkuNok4uiIgMHGxCyYrA4PCCilQ7VAXHiAwknGI7AIBHgme4rLd3AA0pBi7yqZROCMAD79diIaTfUA7477uUAbn7g+Ho84yekACtILAaLyAN6gADa8RX+SHCRvE5TjOkBzgwC7LngAC6v7XvAY6gAAvhe0qNFSVziqgAAidQMDB-5wbeD5PqgH6QTyn6QeSqHWu0KC8ny6DoFKKGUkcGHDL+74UNweEjgR-DcPosC-sCcoeF4CHkVxPGgAATAAjAJQm-qgZDsCgCHBMkXDPg0XiymQqAAMJynunBKgORCQL+67DBcvE3vw0gOfxcLKeM8BiWYOq-pwe4Hti75OKg+JePJmlUax6GGSB2F5sIa4YXFDAAPwBAZqBKiqLljnej7Pk5OWEflqA6jyAB80XGaZiDmZZkBZqlAQfhlwwBNxYjCLBY6AdOs7ztxEGQU5aUtexm7ta4DBdfhPWTn1IEDUuK7DVVwxpYJgiwDqOpAA